### PR TITLE
fix missing 'gi' dependency

### PR DESCRIPTION
**Pull request by hjmodi**
* CLOUDBLD-7559

Signed-off-by: Harsh Modi <hmodi@redhat.com>

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

File: requirements-build.in [hunk 1]
```diff
@@ -1,9 +1,10 @@
-# Generated by pip_find_builddeps.py on Oct 20 2021 09:32:35
+# Generated by pip_find_builddeps.py on Oct 26 2021 19:48:13
 Cython
 cffi>=1.12
 flit_core<4,>=3.2.0
 packaging>=20.0
 poetry-core
+pycairo
```
Comment (chmeliik) [inline requirements-build.in:7]: `pycairo` is a runtime dep of  `PyGObject`, not a builddep

https://gitlab.gnome.org/GNOME/pygobject/-/blob/master/setup.py (it's in install_requires, not setup_requires)

File: test.sh [hunk 2]
```diff
@@ -75,7 +76,7 @@ function setup_osbs() {
   fi
 
   # Pip install packages for atomic-reactor
-  $RUN "${PIP_INST[@]}" -r requirements.txt
+  $RUN "${PIP_INST[@]}" -r requirements.txt --ignore-installed
```
Comment (chmeliik) [inline test.sh:79]: Why `--ignore-installed`? If it's necessary, could you add a comment?

File: .github/workflows/unittests.yaml [hunk 3]
```diff
@@ -57,7 +57,8 @@ jobs:
           COVERALLS_FLAG_NAME: ${{ matrix.os.name }}-${{ matrix.os.version }}-python${{ matrix.os.python }}
           COVERALLS_PARALLEL: true
         run: |
-          pip3 install --upgrade pip
+          sudo apt install python3-testresources
+          pip3 install --upgrade pip==20.0.2
```
Comment (chmeliik) [inline .github/workflows/unittests.yaml:61]: Which of these changes fixes the coveralls issue, and how? Could that be explained in the commit message and/or comment in this file?

File: requirements-build.in (hunk 1 above)
Comment (hjmodi) [reply requirements-build.in:7]: hmm, I see what you're saying and I agree we should move it. But it was the result of the builddep.py script so if we move it we'll have to always move it after running that script.

Comment (chmeliik) [reply requirements-build.in:7]: Did you run the builddeps script before or after pip-compiling the requirements.in? If any runtime dependencies are missing in requirements.txt, that script will mis-identify them as build dependencies

File: test.sh (hunk 2 above)
Comment (hjmodi) [reply test.sh:80]: added comment

File: requirements-build.in (hunk 1 above)
Comment (hjmodi) [reply requirements-build.in:7]: tried your suggestion `pycairo` is in the correct place now :D

**Review by mkosiarc — APPROVED** [review]

File: test.sh (hunk 2 above)
Comment (ben-alkov) [reply test.sh:80]: > it was added due to PyGoObject conflicting with the existing PyGoObject on fedora. I'll add a comment.

When/how did this happen? "It Used To Work"™️

File: requirements-build.in (hunk 1 above)
Comment (chmeliik) [reply requirements-build.in:7]: Why is it still in the build requirements file? Did the find_builddeps script put it there even when it was already in requirements.txt?

File: requirements.in [hunk 4]
```diff
@@ -12,3 +12,5 @@ git+https://github.com/containerbuildsystem/osbs-client@be934592d8c2a90c186bdab6
 cryptography < 3.5
 requests
 koji
+pycairo
```
Comment (chmeliik) [inline requirements.in:15]: Shouldn't have to be declared explicitly, pip-compile should find it.

Comment (hjmodi) [reply requirements.in]: it did. Updated.

builddeps.py still added it in build requirements we will have to remember to remove it.

File: requirements-build.in (hunk 1 above)
Comment (hjmodi) [reply requirements-build.in]: it was added in build requirements even though it was in requirements.txt I had to remove it manually this time.

Comment (chmeliik) [reply requirements-build.in]: Oh, my bad, it is both a runtime and a build time dependency (the build dependency is defined here https://gitlab.gnome.org/GNOME/pygobject/-/blob/master/pyproject.toml#L19). Please leave it in the build requirements, sorry :sweat_smile:

Comment (hjmodi) [reply requirements-build.in:7]: ok, I've added it back to build requirements as well.

**Review by chmeliik — APPROVED** [review]
LGTM 👍

**Pull request merged**